pytorch_lightning/accelerators/tpu_backend.py

```diff
@@ -169,7 +168,7 @@ def to_device(self, batch):
         See Also:
             - :func:`~pytorch_lightning.utilities.apply_func.move_data_to_device`
         """
-        if not XLA_AVAILABLE:
+        if not TPU_AVAILABLE:
             raise MisconfigurationException(
                 'Requested to transfer batch to TPU but XLA is not available.'
                 ' Are you sure this machine has TPUs?'
```

Comment:
should this check be added in Trainer `__init__` itself for both XLA and TPU if `tpu_cores` is requested?? There might be cases when either of them is missing. What do you think??

Reply:
The check for `gpus` is present in `__init__` yes. I guess we could refactor the code here and move the TPU check to trainer `__init__` too. A separate PR for that maybe? :)

Reply:
good, then, pls add TODO comment so we do not forget...